Framework/DataHandling/src/LoadCSNSNexus.cpp

```diff
@@ -0,0 +1,568 @@
+  }
+  return data;
+}
+
+/**
+ * Return modules for special bankName at GPPD
+ * @param[in] bankName :: bank1,bank2,bank3
+ * @return modules
+ */
+vector<string> LoadCSNSNexus::getGPPDModules(string bankName) {
+  vector<string> data;
+  std::map<std::string, std::string> entries;
+  m_file->openGroup(m_entry, "NXentry");
+  m_file->openGroup("instrument", "NXinstrument");
+  entries = m_file->getEntries();
+  if (bankName == "bank3") {
+    for (auto it = entries.begin(); it != entries.end(); ++it) {
+      std::string name = it->first;
+      string tmp = name.substr(0, 7);
+      // if (!name.find("module1") || !name.find("module2"))
+      if (tmp == "module1" || tmp == "module2")
+        data.push_back(name);
+    }
+  } else if (bankName == "bank2") {
+    for (auto it = entries.begin(); it != entries.end(); ++it) {
+      std::string name = it->first;
+      string tmp = name.substr(0, 7);
+      // if (!name.find("module3") || !name.find("module4"))
+      if (tmp == "module3" || tmp == "module4")
+        data.push_back(name);
+    }
+  } else if (bankName == "bank1") {
+    for (auto it = entries.begin(); it != entries.end(); ++it) {
+      std::string name = it->first;
+      string tmp = name.substr(0, 7);
+      // if (!name.find("module5") || !name.find("module6"))
+      if (tmp == "module5" || tmp == "module6")
+        data.push_back(name);
+    }
+  }
+  m_file->closeGroup();
+  m_file->closeGroup();
+  return data;
+}
+
+/**
+ * Judge the input of Bankname
+ * @param[in] inputNames :: Bankname
+ * @return true or false
+ */
+bool LoadCSNSNexus::checkBanknames(const vector<string> &inputNames) {
+  if (inputNames.size() == 0)
+    return false;
+  else if (inputNames.size() > 1) {
+    for (auto it = inputNames.begin(); it != inputNames.end(); it++) {
+      string name = *it;
+      if (name.substr(0, 6) != "module")
+        return false;
+    }
+    return true;
+  } else
+    return true;
+}
+/**
+ * Return modules for special instrument
+ * @param[in] inst :: instrument name
+ * @param[in] inputNames :: Bankname
+ * @return modules
+ */
+vector<string> LoadCSNSNexus::getModules(string inst,
+                                         const vector<string> &inputNames) {
+  vector<string> data;
+  if (inst == "SANS" || inst == "MR")
+    data.push_back("module1");
+  else if (inst == "GPPD") {
+    for (auto it = inputNames.begin(); it != inputNames.end(); it++) {
+      if ((*it).substr(0, 4) == "bank")
+        data = getGPPDModules(*it);
+      else
+        data.push_back(*it);
+    }
+  }
+  return data;
+}
+
+/**
+ * Return pixel id
+ * @param[in] inputList :: moduleList or monitorList
+ * @return pixelId
+ */
+std::vector<int64_t>
+LoadCSNSNexus::getPixelId(const vector<string> &inputList) {
+  std::vector<int64_t> _tmp;
+  std::vector<int64_t> pixelId;
+  m_file->openGroup(m_entry, "NXentry");
+  m_file->openGroup("instrument", "NXinstrument");
+  std::map<std::string, std::string> entries = m_file->getEntries();
```

Comment:
Can we use `for(const auto & [key,value] : entries)` and for other cases later in this file?